pl_examples/loops_customisation/k_fold.py

```diff
@@ -0,0 +1,162 @@
+# Copyright The PyTorch Lightning team.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""
+WARNING: Loop customization is in `pre-alpha release` and the API is likely to change quite a lot !
+Please, open issues with your own particular requests, so the Lightning Team can progressively converge to a great API.
+"""
+
+from dataclasses import dataclass, field
+from typing import Any, Dict, List, Type
+
+import numpy as np
+from sklearn.model_selection import KFold
+from torch.utils.data import Dataset
+
+from pytorch_lightning import _logger as log
+from pytorch_lightning import seed_everything, Trainer
+from pytorch_lightning.callbacks.base import Callback
+from pytorch_lightning.loops.base import ExternalLoop
+from pytorch_lightning.utilities import rank_zero_only
+from pytorch_lightning.utilities.boring_model import BoringDataModule, BoringModel
+from pytorch_lightning.utilities.exceptions import MisconfigurationException
+
+seed_everything(42)
+
+
+class SplitDataset(Dataset):
```

Comment:
what's the difference between this and `torch.utils.data.Subset`?
Can we just replace it as it seems to add no extra functionality

Reply:
I didn't know about it :) Thanks.